### DISCUSSION TITLE
Static vs Dynamic Fields and Dependencies

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#11833] [Help] [answered]
From my experience, multiple dynamic formula fields can significantly slow down the loading of an app. However, the use of static formulas does provide load time benefits in that the calculations have already been done and the result saved in the database. 

BUT, in my specific case, I have formula fields that are dependent upon the values of other formula fields. 

For example, let's say I want users to be able to enter the length, width, and height of a box in either imperial or metric. The user can determine which they are entering with a fourth field, a dropdown. We also want the volume of this box for shipping purposes.

Now, we want to ALWAYS display this data in imperial format. So we also have three additional formula fields (length, width, and height) which are calculated if the original data is metric. Otherwise it's just duplicated. Finally, a fourth formula field takes these dimensions and calculates the volume. 

So we have something like this:

USER INPUT: Dropdown selection (imperial, metric), length, width, height.
FORMULA FIELDS CONVERT IF NECESSARY: Length, Width and Height. 
FORMULA FIELD (Dependent upon the three formula fields above): Volume.

So there is a dependency at the final formula field for volume. I have found the calculation only works when dynamic formulas are used. Static formulas would provide better load times but seem the result in errors. 

My application requires a decent amount of formulas since I'm dealing with product data from US and European sources, as well as financial magic to determine landed costs, per unit pricing, case pricing and on and on. The result is sometimes painful load times. 

Having most of this data calculated in the database is HIGHLY beneficial to the design of the app and any future updates that may be needed but comes at the expense of load times. The alternative is to have calculations done in user view which isn't always possible depending on the component being used.

Any insights into why this is? Is there a better way that I'm not seeing?

**Comment by melohagan** [comment, marked as answer]
> From my experience, multiple dynamic formula fields can significantly slow down the loading of an app. However, the use of static formulas does provide load time benefits in that the calculations have already been done and the result saved in the database.

Definitely agree!

I believe this guide is very close to what you are trying to achieve:
https://docs.budibase.com/docs/calculate-field-value-on-save

In this case no formula columns are used, and the database only stores a standardised unit of measurement, which is probably best practice. 
This is why you typically only store UTC in a database for example, and let the front-end handle the custom unit conversions.
